internal/metrics/oba_rest_api_metrics_test.go

```
@@ -86,3 +93,107 @@ func TestFetchObaAPIMetrics_WithVCR(t *testing.T) {
+	}
+	if strings.Contains(gotURL, apiKey) || strings.Contains(gotURL, "key=") || strings.Contains(gotURL, "user:pass") {
+		t.Fatalf("credential leaked in server_url label %q", gotURL)
+	}
+}
+
+func TestFetchObaAPIMetrics_DoesNotLeakAPIKeyInLogs(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		// #nosec G104
+		w.Write([]byte(`{"code":200,"text":"OK","version":2,"currentTime":123,"data":{"entry":{"agenciesWithCoverageCount":0,"agencyIDs":[]}}}`))
```

golangci-lint (2.12.2)

[error] Error return value of `w.Write` is not checked

(errcheck)